core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala

```diff
@@ -161,10 +162,21 @@ private[spark] abstract class BasePythonRunner[IN, OUT](
             logWarning("Failed to close worker socket", e)
         }
       }
+      if (reuseWorker) {
+        val key = (worker, context.taskAttemptId)
+        PythonRunner.runningMonitorThreads.remove(key)
```

Comment:
Hmmm .. I don't like this very much because we're adding some more possibility for leaking ... this task completion lister itself is not something guaranteed in fact IIRC. 

BTW, it would be great to elabourate what's going on here. So the problem is that there are multiple monitor threads that wait for the same Python (forked) worker with the same socket .. right

Reply:
@HyukjinKwon your comment is addressed in https://github.com/apache/spark/pull/32169/commits/7ed511066f7be831f8ea52ac9f93a3687baecc8d